external/dynamo/generalized/processor.py

```
@@ -0,0 +1,429 @@
+                    )
+                    # Persist per-request metrics to CSV (concurrency-safe)
+                    await self._log_request_metrics(num_tokens=len(all_tokens), latency_ms=latency_ms)
+                    yield {"finish_reason": finish_reason}
+                    return
+
+        except Exception as e:
+            latency_ms = (time.perf_counter() - t0) * 1000.0
+            await self._send_feedback_safely(
+                decision_id, latency_ms, False, len(token_ids), len(all_tokens), "exception"
+            )
+            logger.exception("Engine stream exception: %s", e)
```

Ruff (0.14.8)

Redundant exception object included in `logging.exception` call

(TRY401)